server/libs/infra/src/repositories/media.repository.ts

```diff
@@ -124,4 +126,40 @@ export class MediaRepository implements IMediaRepository {
         .run();
     });
   }
+
+  async extractThumbhash(imagePath: string): Promise<Buffer> {
```

Comment:
Lots of things going on in this function and I'm not sure for what. After creating my own implementation the resulting thumb hashes are the same. I also had to use a base64 encoded value, because there was an error inserting the raw bytes into the database.

```ts
async extractThumbhash(imagePath: string): Promise<string> {
  const maxSize = 100;

  const { data, info } = await sharp(imagePath)
    .resize(maxSize, maxSize, { fit: 'inside', withoutEnlargement: true })
    .raw()
    .ensureAlpha()
    .toBuffer({ resolveWithObject: true });

  const thumbhash = rgbaToThumbHash(info.width, info.height, data);
  return Buffer.from(thumbhash).toString('base64');
}
```

Reply:
Wow that is far more elegant that what I wrote, thanks for that. 
You have any further info on why inserting bytes failed? It seems to work fine on my end.